pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -78,6 +79,16 @@ func NewReconciler(mgr manager.Manager, rateLimiter flowcontrol.RateLimiter) *Re
 	}
 }
 
+func indexCDByClusterPool(o client.Object) []string {
+	cd := o.(*hivev1.ClusterDeployment)
```

Comment:
in case there is some bug and we don't get clusterdeployment, this will panic and stop entire controller.. maybe handle and return empty?

Reply:
IMO that's a bit too paranoid, considering the parameter two before this one in the `IndexField` func is the type in question. Also, none of the other indexer funcs are doing that type check.

https://github.com/openshift/hive/blob/33be31582aa140ce772333d0437eb77cd723cee2/pkg/controller/machinemanagement/machinemanagement_controller.go#L90
https://github.com/openshift/hive/blob/33be31582aa140ce772333d0437eb77cd723cee2/pkg/controller/clusterdeployment/clusterinstalls.go#L227
https://github.com/openshift/hive/blob/33be31582aa140ce772333d0437eb77cd723cee2/pkg/operator/hive/hive_controller.go#L162